fizz-buzz-streams/src/test/java/com/pubsap/eng/streams/FizzBuzzAggregatorTest.java

```diff
@@ -0,0 +1,191 @@
+                .windowedBy(timeWindow)
+                .aggregate(
+                        FizzBuzzAggregator.init,
+                        FizzBuzzAggregator.aggregator,
+                        Materialized.with(inputKeySerde, outputSerde)
+                )
+                .toStream()
+                .map(FizzBuzzMapper.formatOutput)
+                .to("output-topic",producedCounts);
+
+        testDriver = new TopologyTestDriver(builder.build(), properties);
+
+        inputTopic = testDriver.createInputTopic(inputTopicName, inputKeySerde.serializer(), itemSerde.serializer());
+        outputTopic = testDriver.createOutputTopic(outputTopicName, outputKeySerde.deserializer(), outputSerde.deserializer());
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        testDriver.close();
+    }
+
+    @Test
+    public void aggregatorShouldGroupAllTypeOfItems() {
+
+        //Given
+        final List<KeyValue<InputKey, Item>> inputValues = Arrays.asList(
+                new KeyValue<>(new InputKey("client-1"), new Item(ItemValue.Fizz)),
+                new KeyValue<>(new InputKey("client-1"), new Item(ItemValue.Buzz)),
+                new KeyValue<>(new InputKey("client-1"), new Item(ItemValue.FizzBuzz))
+        );
+
+        //When
+        inputTopic.pipeKeyValueList(inputValues);
+
+        //Then
+        final KeyValue<OutputKey,Output> expectedResult = new KeyValue<>(new OutputKey("client-1", "", ""), new Output(1, 1, 1));
+
+        List<KeyValue<OutputKey,Output>> result = outputTopic.readKeyValuesToList();
+
+        final KeyValue<OutputKey,Output> actualResult = result.get(result.size()-1);
+
+        assertEquals(expectedResult.value, actualResult.value);
+        assertEquals(expectedResult.key.getName(), actualResult.key.getName());
+        assertTrue(result.size()==3);
+    }
+
+    @Test
+    public void aggregatorShouldGroupItemBasedOnItemValue() {
+
+        //Given
+        final List<KeyValue<InputKey, Item>> inputValues = Arrays.asList(
+                new KeyValue<>(new InputKey("client-2"), new Item(ItemValue.Fizz)),
+                new KeyValue<>(new InputKey("client-2"), new Item(ItemValue.Fizz)),
+                new KeyValue<>(new InputKey("client-2"), new Item(ItemValue.Buzz)),
+                new KeyValue<>(new InputKey("client-2"), new Item(ItemValue.Buzz)),
+                new KeyValue<>(new InputKey("client-2"), new Item(ItemValue.FizzBuzz)),
+                new KeyValue<>(new InputKey("client-2"), new Item(ItemValue.FizzBuzz))
+        );
+
+        //When
+        inputTopic.pipeKeyValueList(inputValues);
+
+        //Then
+        final KeyValue<OutputKey,Output> expectedResult = new KeyValue<>(new OutputKey("client-2", "", ""), new Output(2, 2, 2));
+
+        List<KeyValue<OutputKey,Output>> actualResultAsList = outputTopic.readKeyValuesToList();
+
+        final KeyValue<OutputKey,Output> actualResult = actualResultAsList.get(actualResultAsList.size()-1);
```

Comment:
if it's about getting the last message, why don't you use the `readKeyValuesToMap`?
Also I think in this case it's actually good to use the list, but just check the entire output.

Reply:
readKeyValuesToMap isn't useful here, the keys have a window start and stop we can't get values by a defined key. Same "problem" for lists, in order to use and compare them we have to transform the List<KeyValue<OutputKey, Output>> to List<KeyValue<String, Output>> result in order to test only client names. I think that's overskill for this test.